resources/sdk/clisdkclient/extensions/config/config.go

```diff
@@ -0,0 +1,89 @@
+package config
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/viper"
+)
+
+type Configuration interface {
+	ProfileName() string
+	Environment() string
+	ClientID() string
+	ClientSecret() string
+	fmt.Stringer
+}
+
+type configuration struct {
+	profileName  string
+	environment  string
+	clientID     string
+	clientSecret string
+}
+
+//ProfileName is the name of the profile being used to run the CLI
+func (c *configuration) ProfileName() string {
+	return c.profileName
+}
+
+//ClientID is the OAuth client id used by the OAuth Client
+func (c *configuration) ClientID() string {
+	return viper.GetString(fmt.Sprintf("%s.client_credentials", c.profileName))
+}
+
+//ClientSecret is the OAuth client secret used by the OAuth Client
+func (c *configuration) ClientSecret() string {
+	return viper.GetString(fmt.Sprintf("%s.client_secret", c.profileName))
+}
+
+//Environment is the Genesys Cloud Environment the CLI will talk to
+func (c *configuration) Environment() string {
+	return viper.GetString(fmt.Sprintf("%s.environment", c.profileName))
+}
+
+func (c *configuration) String() string {
+	return fmt.Sprintf(`{"profileName": "%s", "environment": "%s", "clientName": "%s", "clientSecret": "%s"}`, c.ProfileName(), c.Environment(), c.ClientID(), c.ClientSecret())
+}
+
+//GetConfig retrieves the config for the current profile
+func GetConfig(profileName string) (Configuration, error) {
+	if err := viper.ReadInConfig(); err != nil {
+		return nil, fmt.Errorf("Error reading config file, %s", err)
+	}
+
+	profile := viper.GetViper().Get(profileName)
+
+	if profile == nil {
+		return nil, fmt.Errorf("The profile named %s passed can not be located in the config file.", profileName)
+	}
+
+	return &configuration{profileName: profileName,
+		clientID:     viper.GetString(fmt.Sprintf("%s.client_credentials", profileName)),
+		clientSecret: viper.GetString(fmt.Sprintf("%s.client_secret", profileName)),
+		environment:  viper.GetString(fmt.Sprintf("%s.environment", profileName)),
+	}, nil
+}
+
+func SaveConfig(config Configuration) error {
+	viper.Set(fmt.Sprintf("%s.client_credentials", config.ProfileName()), config.ClientID())
+	viper.Set(fmt.Sprintf("%s.client_secret", config.ProfileName()), config.ClientSecret())
```

Comment:
@carnellj Should we be obfuscating the client secret in some way so it's not stored in plain text?

Reply:
Yes and no.  The problem is that if we obfuscate the code then that means the client must set up all-new credentials only through the Command Line tools.  If the customer ever wants to use this in the context of CI/CD, they would need to manually generate the profiles with the tool and then take the config file and check it into a secret's value or a filesystem.

I am going to chat with info security about this, but frankly, this all runs on the customer's machine and they have complete responsibility to secure these credentials.  We would force obfuscation on any credentials created or used.  It's a good point to at least look into.

Reply:
> that means the client must set up all-new credentials only through the Command Line tools.

Not necessarily. Client secrets are always 43 characters in length. If the obfuscation method will always generate a value that is not 43 characters, then users can put their secret directly in the file and the SDK can know that loading a secret from the config that is exactly 43 characters is a plain value. It can then use it as-is for requests and update the config file to store the obfuscated value for future use.

Reply:
Yeah that sounds clunky.  I put my AWS credentials in the .cli command and they do not perform obfuscation.